test/units/executor/test_task_executor.py

```diff
@@ -230,9 +232,31 @@ def _evaluate_conditional(templar, variables):
         new_items = te._squash_items(items=items, loop_var='item', variables=job_vars)
         self.assertEqual(new_items, ['a', 'b', 'c'])
 
+        mock_task.action = '{{unknown}}'
+        mock_task.args={'name': '{{item}}'}
+        new_items = te._squash_items(items=items, loop_var='item', variables=job_vars)
+        self.assertEqual(new_items, ['a', 'b', 'c'])
+
+        # Maybe should raise an error in this case.  The user would have to specify:
+        # - yum: name="{{ packages[item] }}"
+        #   with_items:
+        #     - ['a', 'b']
+        #     - ['foo', 'bar']
+        # you can't use a list as a dict key so that would probably throw
+        # an error later.  If so, we can throw it now instead.
+        # Squashing in this case would not be intuitive as the user is being
```

Comment:
Squashing is fine in this case, I think, if a list of lists is effectively just another way of writing a list

e.g. sometimes you might want to do:

yum: 
  name="{{item}}"
  state=present
with_items:
- "{{list_of_default_rpms}}"
- "{{list_of_custom_rpms}}"

You can use union to join the lists of course, but I'm not sure what other use case drives this test that isn't amenable to squashing